src/core/txs/wallet.c

```diff
@@ -1,30 +1,46 @@
 #include "wallet.h"
 #include "constants.h"
 #include "crypto.h"
+#include "ser_wallet.h"
 
 #include "mbedtls/ecp.h"
 
 mbedtls_ecdsa_context **build_inputs(Transaction *tx, TxOptions *options) {
   size_t i, num_entries;
   mbedtls_ecdsa_context **keys;
-  WalletPool *map_value, *end_value;
-
+  //WalletPool *map_value, *end_value;
   options->out_total = options->tx_fee;
   for (i = 0; i < options->num_dests; i++)
     options->out_total += options->dests[i].amt;
 
   // Ensure we have enough in wallet before pops
   num_entries = 0;
   options->in_total = 0;
-  map_value = wallet_pool;
-  while (map_value != NULL && options->in_total < options->out_total) {
-    if (map_value->entry->spent == 0) {
+
+  leveldb_readoptions_t *roptions = leveldb_readoptions_create();
+  leveldb_iterator_t *iter = leveldb_create_iterator(wallet_pool_db, roptions);
+
+  for (leveldb_iter_seek_to_first(iter); leveldb_iter_valid(iter); leveldb_iter_next(iter))
+  {
+    size_t key_len, value_len;
+    unsigned const char *key_ptr = (unsigned const char*) leveldb_iter_key(iter, &key_len);
+    unsigned const char *value_ptr = (unsigned const char*) leveldb_iter_value(iter, &value_len);
+
+    WalletEntry *read_wallet_entry = deser_wallet_entry_alloc(NULL, (unsigned char*)value_ptr);
+    if(read_wallet_entry->spent == 0){
       num_entries++;
-      options->in_total += map_value->entry->amt;
+      options->in_total += read_wallet_entry->amt;
+    }
+    free(read_wallet_entry);
+    // free(key_ptr);
+    // free(value_ptr);
```

Comment:
From what I Understand it may actually be returning static memory, I don't know what we should do here, I guess we should gdb and see if it's a heap value...

Reply:
I really don't know...

Reply:
Can confirm that freeing them causes an `invalid pointer` error, so I think we're fine as is. No idea how this is done under the hood though.